core/src/main/scala/akka/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -52,10 +52,10 @@ class JournalQueries(
       .update(true)
 
   private def _highestSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
-    selectAllJournalForPersistenceId(persistenceId).map(_.sequenceNumber).max
+    selectAllJournalForPersistenceId(persistenceId).take(1).map(_.sequenceNumber).max
 
   private def _highestMarkedSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
-    selectAllJournalForPersistenceId(persistenceId).filter(_.deleted === true).map(_.sequenceNumber).max
+    selectAllJournalForPersistenceId(persistenceId).filter(_.deleted === true).take(1).map(_.sequenceNumber).max
```

Comment:
is `.max` needed if the result is already ordered by the id and we take the first?

Reply:
It's not, frankly speaking, I have some troubles with it and I need to educate myself on how to do this in slick in this case.

Reply:
What happens here is that this is a description of the query. 

The query returns `Query[Rep[Long], Long, Seq]`, in Slick parlance it means a query that will return `Seq[Long]`. 

Slick can't know we passed `1` to `take`, that's a runtime value from Slick point of view. It could have been `a` or `num`. Scala knows is hard code, but Slick doesn't. Anyway, moving on...

When we call `.max`, Slick knows that we can only return none or one value. It can then transform the query to a Rep, in that case `Rep[Long]`. 

If Slick had a `takeHead` or `takeFirst` we would be able to replace `take(1).max`, but that's the not case. 

Therefore we need the `.max`. 

Alternatively, we can drop it and change the method signature to ` Query[Rep[Long], Long, Seq]`, but then, when we call it we will have to do:

```scala
 db.run(queries.highestSequenceNrForPersistenceId(persistenceId).result.headOption)
```

I agree that's strange to use `max`, but it's better that this method is self-contained.